### DISCUSSION TITLE
go to v

### DISCUSSION THREAD

**Discussion by rraammiinn** [vlang/v#17452] [Questions and Answers]
can go ecosystem be translated to v entirely or any sort of cooperation with go as zig is doing with c ?

**Comment by JalonSolov** [comment]
[go2v](https://github.com/vlang/go2v) is partially working now.

Other than that, you can use `-b go` to output Go code from V (still WIP), or you can use C interop between Go and V, using Go's `cgo` layer.

**Reply by rraammiinn** [reply]
I mean it becomes drop in replacement to go so it can easily be used to create performant web servers and use all of go's modules as they are v modules  .

**Comment by rraammiinn** [comment]
I think it's possible as v has equivalents for all of go's syntax and support several memory management techniques .

**Reply by rraammiinn** [reply]
so all of source code can be translate to v without any special techniques .

**Reply by JalonSolov** [reply]
Um... no, it can't.  It may seem like that at a quick glance, but no... otherwise it would already be done.

If you want to know why, I suggest you do a more in-depth study of the 2 languages.

**Reply by rraammiinn** [reply]
that's bad .
otherwise it could be go killer .
what about rust ?
it's higher level that rust so is it possible ?

**Reply by rraammiinn** [reply]
but no. 
v and rust are so different .

**Reply by JalonSolov** [reply]
If you spend enough time, and understand both well enough, you can translate anything to anything else.

Some will be easier, some harder, but it can be done.

**Reply by JalonSolov** [reply]
It is similar to translating from one human language to another.  Some are fairly simple (but still have "gotchas"), such as translating between American and British English.  It's much harder to translate between Chinese and English, or Russian and English, but both are done all the time.

**Reply by rraammiinn** [reply]
of course but in general translating a language to a more featured language is easier .
